src/odemis/gui/comp/viewport.py

```diff
@@ -930,6 +930,8 @@ def update_legend_pol_choices(self):
         polarization position is requested.
         """
         choices = self.stream.polarization.choices
+        if len(choices) > 1:
```

Comment:
Let's make it less of an hack, and write:
`if choices == frozenset(POL_POSITIONS):`

BTW, it could be argued that using the metadata strings is still not super user-friendly (although the metadata strings are normally chosen to be user "readable"). It could be possible to have a mapping of the POL_MODE -> nice strings. For instance rhc -> "Right-Hand Circ" (or anything more user friendly but short). Anyway, that'd be for a separate patch.

Reply:
Would you put such a dict (I guess it would be a dict) also in base.py?

Reply:
Yes, I also think that next to POL_POSITIONS would make the most sense. Alternatively, you'd put it somewhere in the GUI.